kie-maven-plugin/src/main/resources/META-INF/plexus/components.xml

```diff
@@ -9,7 +9,16 @@
             <configuration>
                 <phases>
                     <process-resources>org.apache.maven.plugins:maven-resources-plugin:resources,org.kie:kie-maven-plugin:validateDMN</process-resources>
-                    <compile>org.apache.maven.plugins:maven-compiler-plugin:compile,org.kie:kie-maven-plugin:generateModel,org.kie:kie-maven-plugin:generateDMNModel,org.kie:kie-maven-plugin:generatePMMLModel,org.apache.maven.plugins:maven-compiler-plugin:compile,org.kie:kie-maven-plugin:build</compile>
+                    <compile>
+                        org.apache.maven.plugins:maven-compiler-plugin:compile,
+                        org.kie:kie-maven-plugin:generateModel,
+                        org.kie:kie-maven-plugin:generateDMNModel,
+                        org.kie:kie-maven-plugin:generatePMMLModel,
+                        org.apache.maven.plugins:maven-compiler-plugin:compile,
+                        org.kie:kie-maven-plugin:build,
+                        org.kie:kie-maven-plugin:generateANC,
+                        org.apache.maven.plugins:maven-compiler-plugin:compile
```

Comment:
Are you sure `build` is not the last one?
Btw I think we should review this mechanism, now we have three maven:compile steps :thinking:

Reply:
I know, but unfortunately ANC compiler needs the rete and the rete is only created at runtime. Then it has to generate those classes and compile them.